### PR TITLE
Refactoring Soundcloud data access methods to a different class

### PR DESCRIPTION
**Pull request by VegaFromLyra**
### Refactor `MainViewController`
- Moved most of the Sound cloud specific code out of this controller into separate model and utility classes
- The MainViewController now talks to `MusicSource` to get a track and then play it
### Models
- `MusicSource` contains the logic to fetch a random track from SoundCloud each time the user launches the app and/or clicks next on the player
- `Track` encapsulates information about a single track
### Helper classes
- Currently we only have a single `Utility` class that has a helper method
### Other changes
- Changes formatting of methods so that function opening braces are on the same line as the function name
- While testing I did not encounter a problem where a stream url can be null so throwing an exception when that happens to catch that problem

### CODE REVIEW

File: SCloudRandomizer/MainViewController.m [hunk 1]
```diff
@@ -7,24 +7,34 @@
 //
 
 #import "MainViewController.h"
+#import "Utility.h"
+#import "Track.h"
 
+const double LoggedOutBackgroundImage_Opacity = 0.7;
```
Comment (akshay23) [inline SCloudRandomizer/MainViewController.m]: Its good objective-C practice to add the 'static' keyword in front of the constants. And remove underscore.

File: SCloudRandomizer/MainViewController.m [hunk 2]
```diff
@@ -474,33 +335,19 @@ - (void)setupUI:(NSDictionary *)track
...
-    {
-        urlString = [[[track objectForKey:@"user"] objectForKey:@"avatar_url"] stringByReplacingOccurrencesOfString:@"-large" withString:@"-t300x300"];
-    }
-    else
-    {
-        urlString = [(NSString *)albumArt stringByReplacingOccurrencesOfString:@"-large" withString:@"-t300x300"];
-    }
-    
-    NSURL *imgUrl = [NSURL URLWithString:urlString];
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
```
Comment (akshay23) [inline SCloudRandomizer/MainViewController.m]: Now that we have a Utils class, we should add a class method that returns a dispatch_get_global_queue. To do this you could just add a new class method to Utils. Something like:

 `+ (dispatch_queue_t)getGlobalBackgroundQueue;`

And in the m file the details would be:

`+ (dispatch_queue_t)getGlobalBackgroundQueue`
{
    return dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0);
}

ANd then replace all the places we use the thread with getGlobalBackgroundQueue

File: SCloudRandomizer/MainViewController.m [hunk 3]
```diff
@@ -7,24 +7,34 @@
...
+
 @property BOOL isCurrentSongLiked;
 @property NSUInteger currentSongNumber;
-@property (strong, nonatomic) NSDictionary *currentTrack;
-@property (strong, nonatomic) SearchParams *mySearchParams;
+@property (strong, nonatomic) Track *currentTrack;
+@property (strong, nonatomic) SearchParams *searchParams;
+@property MusicSource *musicSource;
+
+- (void)getNextTrack:(singleTrackDownloaded)completionHandler;
```
Comment (akshay23) [inline SCloudRandomizer/MainViewController.m]: Dont need to add this private method in the interface

File: SCloudRandomizer/Utility.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//  SCloudRandomizer
+//
+//  Created by Asha Balasubramaniam on 4/30/15.
+//  Copyright (c) 2015 Akshay Bharath. All rights reserved.
+//
+
+@interface Utility : NSObject
+
++ (NSString *)convertFromMilliseconds:(long)duration;
+
```
Comment (akshay23) [inline SCloudRandomizer/Utility.h:14]: Move the stringIsNilOrEmpty function from GlobalData.h to Utils

File: SCloudRandomizer/MusicSource.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
+//
+//  MusicSource.h
+//  SCloudRandomizer
+//
+//  Created by Asha Balasubramaniam on 4/29/15.
+//  Copyright (c) 2015 Akshay Bharath. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+@class Track;
```
Comment (akshay23) [inline SCloudRandomizer/MusicSource.h:11]: Leave a line as space between this an import line

File: SCloudRandomizer/MusicSource.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+@interface MusicSource: NSObject
+
+
```
Comment (akshay23) [inline SCloudRandomizer/MusicSource.h]: nit: you have two new lines, lets keep it at one

**Comment by akshay23** [comment]
Looks good

File: SCloudRandomizer/MainViewController.m (hunk 1 above)
Comment (VegaFromLyra) [inline SCloudRandomizer/MainViewController.m]: Done

File: SCloudRandomizer/MainViewController.m (hunk 2 above)
Comment (VegaFromLyra) [inline SCloudRandomizer/MainViewController.m]: Good call, updated

File: SCloudRandomizer/MainViewController.m (hunk 3 above)
Comment (VegaFromLyra) [inline SCloudRandomizer/MainViewController.m]: Removed

File: SCloudRandomizer/Utility.h (hunk 4 above)
Comment (VegaFromLyra) [inline SCloudRandomizer/Utility.h:14]: Done

File: SCloudRandomizer/MusicSource.h (hunk 5 above)
Comment (VegaFromLyra) [inline SCloudRandomizer/MusicSource.h:11]: Done

File: SCloudRandomizer/MusicSource.h (hunk 6 above)
Comment (VegaFromLyra) [inline SCloudRandomizer/MusicSource.h]: Updated

**Pull request merged**